### DISCUSSION TITLE
Multi-level House and Clusters

### DISCUSSION THREAD

**Discussion by adlawton** [mKeRix/room-assistant#890] [Q&A]
Has anyone else found a reliable way to track between floors on a multi-level house? I set maxDistance on BLE which helps a lot but where I run into issues is when I have conflicting signals between floors. For instance, my Bedroom is directly over my Dining Room so my cat will flicker between my Bedroom and Dining Room because they're equi-distant from both nodes. Right now I have 9 RPi Zero's spread between 3 floors, but they're all on a single cluster. Should I be utilizing Clusters differently? Is there ever a good use case to have more than one Cluster? Would somehow breaking my 9 RPi's into 3 Clusters (one per floor) somehow allow it to differentiate what floor its on? I've been fiddling around with this on and off for a month or two but any advice would be appreciated.

**Comment by dawgfan715** [comment]
Great question.  I am dealing with the same issues right now.

**Comment by jmutnick** [comment]
Having the same issue.  I wonder if reducing the minRssi value would help.  Any thoughts?

**Comment by andreasgily** [comment]
Uh. this one could be tricky. Maybe measuring the distance between the cluster nodes in relation to the BLE device could be the key, but you would have to position all your cluster nodes in a grid-like structure? Or maybe tagging the cluster nodes with a "floor" group and if the BLE device changes floor group it kind of sticks there until coming close to a cluster node near to stairs or sth.

**Reply by srvoets** [reply]
I am barely following what you are saying, but somewhat understand.

I have one device in my basement, 2 on main floor, and 2 on 2nd floor, many of them are "above" each other and I can be in the living room and RA says im in the office (above me)

**Comment by dawgfan715** [comment]
By any chance is there an attic above the upstairs room?  I have the same
issue and I don’t have an attic.  I thought about putting a sensor in the
ceiling upstairs and playing with the Rssi settings.  For now I removed
mine from upstairs.

Another idea I had was to use an additional tag or beacon  for upstairs and
put it in my wallet and keep range real small.    In short I’m still
looking at ideas
